index.js

```diff
@@ -21,20 +21,19 @@ var PGPubsub = function (conString, options) {
   this.conFails = 0;
 
   options = options || {};
+  options.retries = options.retries || 9;
 
   this.retry = new Retry({
     name: 'pubsub',
     try: function () {
       var db = new pg.Client(self.conString);
-      db.on('error', function () {
-        self.retry.reset();
-        if (self.channels.length) {
-          self.retry.try();
-        }
-      });
       return new Promise(function (resolve, reject) {
+        db.on('error', reject);
```

Comment:
Some other early feedback: This wouldn't work if there's a database error _after_ the initial connect as the Promise has already been resolved by then and a Promise can only be resolved or rejected once.

The current error handler is meant to catch errors that happens after a successful connect so that if the database somehow eg. shuts down or otherwise closes the connection, that a new connection can automatically be established the next time one is needed.

Any solution would have to take that scenario into account as well.